controller/controller_test.go

```diff
@@ -825,6 +832,141 @@ func TestStartAndShutdown(t *testing.T) {
 	}
 }
 
+type CountingLeaderAwareReconciler struct {
+	reconciler.LeaderAwareFuncs
+
+	m     sync.Mutex
+	Count int
+}
+
+var _ reconciler.LeaderAware = (*CountingLeaderAwareReconciler)(nil)
+
+func (cr *CountingLeaderAwareReconciler) Reconcile(ctx context.Context, key string) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	if cr.IsLeaderFor(types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}) {
+		cr.m.Lock()
+		defer cr.m.Unlock()
+		cr.Count++
+	}
+	return nil
+}
+
+func TestStartAndShutdownWithLeaderAwareNoElection(t *testing.T) {
+	defer ClearAll()
```

Comment:
```suggestion
	t.Cleanup(ClearAll)
```
👅 
So in other places.